examples/router_resource.rs

```diff
@@ -0,0 +1,65 @@
+//! Example: Updating components with use_resource
+//! -----------------
+//!
+//! This example shows how to use use_reactive to update a component properly
+//! when linking to it from the same component, when using use_resource
+
+use dioxus::prelude::*;
+
+#[derive(Clone, Routable, Debug, PartialEq)]
+enum Route {
+    #[route("/")]
+    Home {},
+    #[route("/blog/:id")]
+    Blog { id: i32 },
+}
+
+fn main() {
+    launch(App);
+}
+
+#[component]
+fn App() -> Element {
+    rsx! {
+        Router::<Route> {}
+    }
+}
+
+#[component]
+fn Blog(id: ReadOnlySignal<i32>) -> Element {
+    async fn future(n: i32) -> i32 {
+        n
+    }
+
+    // if you use the naive approach, the "Blog post {id}" below will never update when clicking links!
+    // let res = use_resource(move || future(id));
+
+    // the use_reactive hook is required to properly update when clicking links to this component, from this component
```

Comment:
You merged this with comments that no longer apply because you switched to readonlysignal

Reply:
Oops, fixed in 77b08f2330ebf6ad750adbcfd9f28967509765d4

Reply:
Thanks! The header also mentions use_reactive, and honestly I wonder if using both use_reactive and readonlysignal in this example would help guide people to when to use one vs the other. Or is there a different example we should point at in the header for use_reactive, for comparison?

Reply:
The [use_reactive docs](https://github.com/DioxusLabs/dioxus/blob/77b08f2330ebf6ad750adbcfd9f28967509765d4/packages/hooks/docs/use_resource.md?plain=1#L112-L141) also show how to use both ReadOnlySignals and `use_reactive` with `use_resource`. Any time you can turn your value into a reactive value with readonlysignal, I would recommend that approach. If you have a raw rust value that you can't make reactive, you can use `use_reactive` instead